### DISCUSSION TITLE
Border around the Ulauncher window

### DISCUSSION THREAD

**Discussion by friday** [Ulauncher/Ulauncher#1349] [Troubleshooting]
There are several factors that can lead to this issue appearing, so the first thing you need to do is to understand the problem. If you get a thick black border around the Ulauncher window, it's probably because your window manager does not support compositing, and the border is actually how your window manager renders the shadow around the Ulauncher window.

While if you get a thin border and/or a title bar, this is because your window manager does not respect WM hints to hide the title bar.

## Disabling shadows for compositing window manager
If you don't use a compositing window manager, then Ulauncher's shadows will look like big black box or similar on Ulauncher 5 and older. To avoid this you need to start ulauncher with a command line flag: `ulauncher --no-window-shadow`.

Ulauncher version 6 (beta) has removed this flag in favor of trying to detect and disable shadows when the compositor doesn't support them. This behavior has been confirmed to work on i3.

## Window manager specific workarounds to hide borders

Window managers and DEs can add their own borders by default that you have to override. See examples how to do this for specific DEs:

### KDE Plasma
Most likely it will just work if you use Ulauncher v6 and have `gir1.2-gtklayershell-0.1` installed. If this does not work, see [dedicated page](https://github.com/Ulauncher/Ulauncher/discussions/1501) for details.

### Hyprland
Hyprland handles window borders according to your `hyprland.conf` configuration. You can add exceptions to those rules specifically for Ulauncher. For more on these config options:

Example [window config](https://wiki.hypr.land/Configuring/Window-Rules/):

```
windowrule {
    name = ulauncher
    match:title = negative:Ulauncher Preferences
    match:class = ulauncher
    border_size = 1
    border_color = rgb(ff0000)
}
```

Example [layer config](https://wiki.hypr.land/Configuring/Window-Rules/#layer-rules):
```
layerrule {
    name = ulauncher
    match:namespace = ulauncher
    blur = on
    ignore_alpha = 0.3
    animation = easeInOutCubic
}
```

Credit to [SapphirusBeryl](https://github.com/SapphirusBeryl)

### Sway
You have to tell sway not to put borders on the ulauncher window. Example sway config:

```
exec GDK_BACKEND=wayland ulauncher --hide-window
for_window [app_id="ulauncher"] floating enable, border none
```

Credit to [comment by evilhamsterman](https://github.com/Ulauncher/Ulauncher/issues/230#issuecomment-570736422)

### Picom
```py
blur-background-exclude = [
  # others
  "class_g  = 'Ulauncher'"
]
```

Credit to [comment by twnaing](https://github.com/Ulauncher/Ulauncher/issues/262#issuecomment-1257831153)

### BSPWM
```sh
bspc rule -a Ulauncher state=floating border=off
```

Credit to this [Reddit post](https://www.reddit.com/r/bspwm/comments/zw7451/comment/j1u8t9i/)

### Compton
```sh
compton --shadow-exclude 'class_g = Ulauncher'
```

See [this Compton issue](https://github.com/chjj/compton/issues/182) for more details

Credit to [comment by subdavis](https://github.com/Ulauncher/Ulauncher/issues/262#issuecomment-524141296)

**Comment by shilkpatel** [comment]
Hi
I have the black border around the launcher.

when I run ulauncher with --no-window-shadow, it goes away.

but when I tried to run the systemctl command to make it run on startup it didn't work so i used the gui version, it does start on startup but still has the black border.

How would I run the program with the --no-window-shadow flag on startup?

**Reply by friday** [reply]
Edit the systemd unit (likely /usr/share/applications/io.ulauncher.Ulauncher.desktop) and run `systemctl --user daemon-reload`, or use the v6 beta since it handles this without needing any flag.

**Reply by shilkpatel** [reply]
how would i download the v6 as the other methods haven't worked

**Reply by friday** [reply]
What distro/package manager are you on?

**Reply by shilkpatel** [reply]
Endeavour OS/ using yay

**Reply by friday** [reply]
`yay -S ulancher-git`

**Reply by shilkpatel** [reply]
the beta version doesn't have the black border. Thanks

But it didn't automatically create the desktop shortcut, as it said that it doesn't support the shortcut for the desktop environment. I use lxqt.

I have manually added it so it all works out in the end.
